File: ios/WhoCalled/Shared/ReportCategory.swift
import Foundation

/// Predefined report categories (no free text → safe for RGPD, no moderation).
enum ReportCategory: String, CaseIterable, Identifiable {
  case telemarketing
  case scam
  case robocall
  case silent
  case debt
  case survey
  case other = "unknown"

  var id: String { rawValue }

  var label: String {
    switch self {
    case .telemarketing: return "Démarchage"
    case .scam: return "Arnaque"
    case .robocall: return "Appel automatisé"
    case .silent: return "Appel silencieux"
    case .debt: return "Recouvrement"
    case .survey: return "Sondage"
    case .other: return "Autre"
    }
  }

  static func from(_ api: String?) -> ReportCategory? {
    guard let api else { return nil }
    return ReportCategory(rawValue: api)
  }
}

File: android/app/src/main/java/com/whocalled/android/data/ReportCategory.kt
package com.whocalled.android.data

/**
 * Predefined report categories (no free text → safe for RGPD, no moderation).
 * `api` is the value sent to the backend; `label` is shown in the UI.
 */
enum class ReportCategory(val api: String, val label: String) {
    TELEMARKETING("telemarketing", "Démarchage"),
    SCAM("scam", "Arnaque"),
    ROBOCALL("robocall", "Appel automatisé"),
    SILENT("silent", "Appel silencieux"),
    DEBT("debt", "Recouvrement"),
    SURVEY("survey", "Sondage"),
    OTHER("unknown", "Autre");

    companion object {
        fun fromApi(value: String?): ReportCategory? =
            entries.firstOrNull { it.api == value }
    }
}
